popup.js

```diff
@@ -237,8 +237,12 @@ function handleButtons () {
 function fillExpTable() {
   /* Fills expTable with numbers according to a certain formula.
    */
-  for (var i = 0; i < maxLevel; i++) {
-    expTable[i] = Math.floor(i * i / 2);
+  expTable[1] = 0;
+  for (var i = 2; i < maxLevel; i++) {
+    expTable[i] = expTable[i - 1] + (4 + (i - 1) * (Math.log10(i - 1) + 1));
```

Comment:
Why are you using 2 loops?
```
expTable[i] = Math.floor(expTable[i - 4] + (4 + (i - 1) * (Math.log10(i - 1) + 1)));
```

Reply:
2 loops are because I want to have values closer to your idea and rounding it before calculating next value may do some damage to the algorithm. Another idea is to round "i-1" element.

Reply:
Ok fine. Merge it after solving the conflicts.